### ISSUE TITLE
Google Drive change incorrect mime-type of file

### ISSUE THREAD

**Issue by Sparoney** [rclone/rclone#4060]
#### What is your current rclone version (output from `rclone version`)?

v1.51.0-089-gdfc7215b-beta

#### What problem are you are trying to solve?

I have some files in Google Drive that were incorrectly uploaded. They are mkv files however Google thinks they are Binary files, and thus doesn't show a video preview. They can be played fine with rclone mount.

#### How do you think rclone should be changed to solve that?

Add an option in rclone copy to force a mime-type change based on the file extension. For example a --force-mime-type flag or similar. This will do the following:

Current behaviour: Making a server side copy of the .mkv file with the (incorrect) Binary mime-type will create a copy with the same (incorrect) Binary mime-type
New behaviour with flag: Making a server side copy of the .mkv file with the (incorrect) Binary mime-type will create a copy with the correct mime-type determined by the .mkv extension.

## 

I hope this makes sense, if it doesn't I can explain further.

Thanks!

**Comment by darthShadow** [comment]
This was how it used to work earlier with server-side copies also refreshing the mime-type. It was removed because it always used to time out from Google when the mime-type was being changed. That's probably because it involves a full copy rather than a simple link or something similar.

**Comment by Sparoney** [comment]
Yes you're right actually that reminds me. I went back through the commits to where the mime-type code was changed and tried using it the old way. The copy would just be stuck on 0/0 the whole time. I'm guessing this can't be fixed/implemented

**Comment by darthShadow** [comment]
You can obviously do an upload/download of the affected files to fix the mime-types but I doubt anything can be done about server-side copies unless Google fixes it on their backend.

**Comment by Sparoney** [comment]
> You can obviously do an upload/download of the affected files to fix the mime-types but I doubt anything can be done about server-side copies unless Google fixes it on their backend.

Yeah I was hoping I wouldn't need to download and reupload because my upload speed is trash but I'll just try and get a virtual machine.

**Comment by darthShadow** [comment]
You could use VM's from Google Cloud. Their upload and download won't be counted as long as it's from and to Google Drive. If you can make do with the instance available in the free tier, then this whole operation will be basically free.

**Comment by Dibbyo456** [comment]
@Sparoney I'm having a hard time understanding what you mean. Can you explain it a little more?
Rclone supposed to do exact clone, isn't it?

**Comment by Sparoney** [comment]
> @Sparoney I'm having a hard time understanding what you mean. Can you explain it a little more?
> Rclone supposed to do exact clone, isn't it?

Basically, I had copied a bunch of .mkv files from another person's drive. I'm not sure how they were uploaded, but they were uploaded in a way that Google doesn't recognise the file format and thinks its a "Binary file". This means instead of having the red video icon and being able to watch the video on the drive, it has the blue page icon and can't show a preview.

The videos can be played perfectly fine from an rclone mount. 

I was wondering if there was a way to when making a new server-side copy, change the mime-type to the correct one based on the file extension. @darthShadow was saying this used to be the original behaviour but was changed due it timing out and not working.

**Comment by Dibbyo456** [comment]
Ah, understood. I think they best way to deal with this scenario would be fix those files mime-types in the original copy. You can do this with [Drive API](https://developers.google.com/drive/api/v3/reference/files/update) but sadly no way do this with Rclone.
